File: C/09_Math2/BOJ_11_1002_터렛.c
#include <stdio.h>
#include <math.h>
main() {
	int t, cnt = 0;
	scanf("%d", &t);
	int* result = (int*)malloc(sizeof(int) * t);
	while (t--) {
		int x1, y1, r1, x2, y2, r2;
		scanf("%d %d %d %d %d %d", &x1, &y1, &r1, &x2, &y2, &r2);
		double d = sqrt(pow(x1 - x2, 2) + pow(y1 - y2, 2));
		if (d == 0) {
			if (r1 == r2) {
				result[cnt++] = -1;
			}
			else {
				result[cnt++] = 0;
			}
		}
		else if (d == r1 + r2 || d == abs(r1 - r2)) {
			result[cnt++] = 1;
		}
		else if (d < r1 + r2 && d > abs(r1 - r2)) {
			result[cnt++] = 2;
		}
		else {
			result[cnt++] = 0;
		}
	}
	for (int i = 0; i < cnt; i++) {
		printf("%d\n", result[i]);
	}
}

File: JAVA/09_Math2/BOJ_11_1002_터렛.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int t = Integer.parseInt(br.readLine());
		StringBuilder sb = new StringBuilder();
		while (t-- > 0) {
			StringTokenizer st = new StringTokenizer(br.readLine(), " ");
			int x1 = Integer.parseInt(st.nextToken());
			int y1 = Integer.parseInt(st.nextToken());
			int r1 = Integer.parseInt(st.nextToken());
			int x2 = Integer.parseInt(st.nextToken());
			int y2 = Integer.parseInt(st.nextToken());
			int r2 = Integer.parseInt(st.nextToken());
			double d = Math.sqrt(Math.pow(x1 - x2, 2) + Math.pow(y1 - y2, 2));
			if (d == 0) {
				if (r1 == r2) {
					sb.append(-1).append("\n");
				} else {
					sb.append(0).append("\n");
				}
			} else if (d == r1 + r2 || d == Math.abs(r1 - r2)) {
				sb.append(1).append("\n");
			} else if (d < r1 + r2 && d > Math.abs(r1 - r2)) {
				sb.append(2).append("\n");
			} else {
				sb.append(0).append("\n");
			}
		}
		System.out.print(sb);
	}
}